process/transaction/shardProcess.go

```diff
@@ -377,6 +377,11 @@ func (txProc *txProcessor) processTxFee(
 
 	if isUserTxOfRelayed {
 		totalCost := txProc.economicsFee.ComputeFeeForProcessing(tx, tx.GasLimit)
+		shouldConsiderMoveBalanceFee := dstShardTxType == process.MoveBalance &&
+			txProc.enableEpochsHandler.IsFixRelayedMoveBalanceFlagEnabled()
+		if shouldConsiderMoveBalanceFee {
+			totalCost = txProc.economicsFee.ComputeMoveBalanceFee(tx)
```

Comment:
this is not correct. Rewriting the totalCost to moveBalance - will only add the move balance part, and it will not use the gasLimit of the transaction. So it is only 50K + 1.5K*len(txData). You need to use ComputeTxFee(tx data.TransactionWithFeeHandler) *big.Int function. This handles correctly. Do this on all this implementation

Reply:
pushed